x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -198,29 +198,15 @@ func (m *MongoCrypt) CreateExplicitEncryptionContext(doc bsoncore.Document, opts
 	algoStr := C.CString(opts.Algorithm)
 	defer C.free(unsafe.Pointer(algoStr))
 
-	switch opts.Algorithm {
-	case "Indexed":
-		if ok := C.mongocrypt_ctx_setopt_index_type(ctx.wrapped, IndexTypeIndexed); !ok {
-			return nil, ctx.createErrorFromStatus()
-		}
-	case "Unindexed":
-		if ok := C.mongocrypt_ctx_setopt_index_type(ctx.wrapped, IndexTypeUnindexed); !ok {
-			return nil, ctx.createErrorFromStatus()
-		}
-	default:
-		if ok := C.mongocrypt_ctx_setopt_algorithm(ctx.wrapped, algoStr, -1); !ok {
-			return nil, ctx.createErrorFromStatus()
-		}
+	if ok := C.mongocrypt_ctx_setopt_algorithm(ctx.wrapped, algoStr, -1); !ok {
+		return nil, ctx.createErrorFromStatus()
 	}
 
-	if opts.QueryType != nil {
-		switch *opts.QueryType {
-		case options.QueryTypeEquality:
-			if ok := C.mongocrypt_ctx_setopt_query_type(ctx.wrapped, 1); !ok {
-				return nil, ctx.createErrorFromStatus()
-			}
-		default:
-			return nil, fmt.Errorf("unsupported value for QueryType: %v", opts.QueryType)
+	if opts.QueryType != "" {
+		queryStr := C.CString(opts.QueryType)
+		defer C.free(unsafe.Pointer(queryStr))
+		if ok := C.mongocrypt_ctx_setopt_query_type(ctx.wrapped, queryStr, -1); !ok {
```

Comment:
Is it safe to free the `queryStr` variable after calling `mongocrypt_ctx_setopt_query_type`? If so, there are a few places in this file where we create a `CString` and don't free it (e.g. in `NewMongoCrypt`). Should we always free `CString`s?

Reply:
I cannot find a canonical reference.

https://pkg.go.dev/cmd/cgo shows this example:

```go
	cs := C.CString("Hello from stdio")
	C.myprint(cs)
	C.free(unsafe.Pointer(cs))
```

https://zchee.github.io/golang-wiki/cgo/ notes:

> One important thing to remember is that C.CString() will allocate a new string of the appropriate length, and return it. That means the C string is not going to be garbage collected and it is up to you to free it.

Updated other uses of CString to free.